File: 2044-count-number-of-maximum-bitwise-or-subsets/2044-count-number-of-maximum-bitwise-or-subsets.cpp
class Solution {
public:
    
    int helper(int indx,int orr,int ans,vector<int>&nums)
    {
        if(indx==nums.size())
        {
            return orr==ans;
        }
        return helper(indx+1,orr,ans,nums)+helper(indx+1,orr|(nums[indx]),ans,nums);
    }
    
    int countMaxOrSubsets(vector<int>& nums) {
        int ans=0;
        for(int i=0;i<nums.size();i++)ans|=nums[i];
        return helper(0,0,ans,nums);
    }
};

File: 2044-count-number-of-maximum-bitwise-or-subsets/2044-count-number-of-maximum-bitwise-or-subsets.java
class Solution {

    public int countMaxOrSubsets(int[] nums) {
        int maxOr=0,size=nums.length;
        for(int i=0;i<size;i++)
            maxOr|=nums[i];
        
        int subset=0,possible=(1<<size);
        for(int i=0;i<possible;i++)
        {
            int orr=0;
            for(int j=0;j<size;j++)
            {
                if((i&(1<<j))>0)
                {
                    orr|=nums[j];
                }
            }
            subset+=(maxOr==orr?1:0);
        }
        return subset;
    }
}